### PR TITLE
MBS-4555, MBS-11092: Switch to SameSite=Strict for login cookies and preserve POST parameters on login

### PR DESCRIPTION
**Pull request by mwiencek**
https://tickets.metabrainz.org/browse/MBS-4555
https://tickets.metabrainz.org/browse/MBS-11092

This'd be useful to put on test, since it has an SSL cert (though I did test it with my personal domain already).

### CODE REVIEW

**Review by reosarevok — COMMENTED** [review]
The idea seems good but it's hard for me to imagine how the seeding page flow works. I'll throw this on test, think we're not testing anything there ATM.

**Review by reosarevok — APPROVED** [review]
Tested it and it seems fine to me. We might want to add some info on why this is necessary in the "Confirm Form Submission" page though.

Under the line
`"You are about to submit a request to https://test.musicbrainz.org/release/add originating from https://sounds.pear.com. Continue?"
`
I'd suggest having something like 

```
"This confirmation is important to ensure no malicious actor can use your account to modify the data without your knowledge.

Below this line you can review the data being sent and make any modifications if desired:"
```

Or something to that effect. Your wording is likely to be better than mine.

Re: log in cookie, I guess that doesn't store the username so we could display it on the header?

File: lib/MusicBrainz/Server/Controller/Admin.pm [hunk 1]
```diff
@@ -90,9 +90,9 @@ sub delete_user : Path('/admin/user/delete') Args(1) RequireAuth HiddenOnSlaves
     my ($self, $c, $name) = @_;
 
     my $editor = $c->model('Editor')->get_by_name($name);
-    $c->stash->{viewing_own_profile} = $c->user_exists && $c->user->id == $editor->id;
+    $c->detach('/user/not_found') if !$editor || $editor->deleted;
 
-    $c->detach('/error_404') if !$editor || $editor->deleted;
+    $c->stash->{viewing_own_profile} = $c->user_exists && $c->user->id == $editor->id;
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply lib/MusicBrainz/Server/Controller/Admin.pm:95]: It's not that `$editor->id` was undef, but that `$editor` was -- you can't dereference undef.

**Comment by mwiencek** [comment]
Firstly, thanks for putting this on test for me. I planned to do it today with some instructions but you already figured everything out. :) I'll look into adding that message.

> Re: log in cookie, I guess that doesn't store the username so we could display it on the header?

`remember_login` does store the name, but we don't have access to that cookie there (otherwise we wouldn't be logged out). Adding a new third party cookie for just the name seemed wrong; I was almost inclined to just make the page simpler altogether by removing the headers/footers. Though like I said, it might also be fine how it is.

**Comment by reosarevok** [comment]
> I was almost inclined to just make the page simpler altogether by removing the headers/footers.

I'd be fine with that as long as we leave enough of the design to make it clear it's a legit part of MB and not some weird phishing clone.

**Review by yvanzo — APPROVED** [review]
It’s impressive! Should we send a special warning to userscript authors about these changes?

**Comment by mwiencek** [comment]
> Should we send a special warning to userscript authors about these changes?

I don't think it'll help much, since it only affects importers and there's nothing they need to fix. (Userscripts that run on the same origin, which are most, can still make use of the samesite=strict cookies -- I tested most of the popular ones.)

**Comment by jesus2099** [comment]
Nicely done with its request parameter display and everything!
I tested #1788 with `seed-love-bug.html` and Picard’s **Add Cluster As Release…** plugin, to beta, and it worked good! :)

**Pull request merged**